Backpack-SwiftUI/Calendar/Classes/Single/SingleCalendarContainer.swift

```diff
@@ -30,20 +30,36 @@ struct SingleCalendarContainer<MonthHeader: View, DayAccessoryView: View>: View
     @ViewBuilder
     private func makeDayCell(_ dayDate: Date) -> some View {
         CalendarSelectableCell {
-            if selection == dayDate {
-                SingleSelectedCell(calendar: calendar, date: dayDate)
-            } else {
+            switch selection {
+            case .single(let date):
+                if date == dayDate {
+                    SingleSelectedCell(calendar: calendar, date: dayDate)
+                } else {
+                    DefaultCalendarDayCell(calendar: calendar, date: dayDate)
```

Comment:
if you appliy the above comment about optionality, I think this if let would be done on the selection itself, rather than the date.
```swift
switch selection {
  case single(let date): SingleSelectedCel(...)
  case wholeMonth(..): ....
  case nil: DefaultCalendarDayCell(...)
}

Reply:
I removed the optional, but we still need this statement to determine which date needs to be highlighted among other dates when a date has been selected `date == dayDate`. 

When the tap action happens, we change the state, and the change of state helps us to render the changes.